packages/runtime/runtime-utils/src/summarizerNode/summarizerNodeWithGc.ts

```diff
@@ -478,6 +478,41 @@ class SummarizerNodeWithGC extends SummarizerNode implements IRootSummarizerNode
 		return child;
 	}
 
+	/**
+	 * Updates the state of the child if required. For example, if a summary is currently being  tracked, the child's
+	 * summary tracking state needs to be updated too.
+	 * Also, in case a child node gets realized in between Summary Op and Summary Ack, let's initialize the child's
+	 * pending summary as well. Finally, if the pendingSummaries entries have serializedRoutes, replicate them to the
+	 * pendingSummaries from the child nodes.
+	 * @param child - The child node whose state is to be updated.
+	 */
+	protected maybeUpdateChildState(child: SummarizerNodeWithGC) {
+		// If a summary is in progress, this child was created after the summary started. So, we need to update the
+		// child's summary state as well.
+		if (this.isSummaryInProgress()) {
+			child.wipReferenceSequenceNumber = this.wipReferenceSequenceNumber;
+		}
+		// In case we have pending summaries on the parent, let's initialize it on the child.
+		if (child.latestSummary !== undefined) {
+			for (const [key, value] of this.pendingSummaries.entries()) {
+				const serializedRoutes = (value as SummaryNodeWithGC).serializedUsedRoutes;
```

Comment:
This is not correct. The child's serialized routes will be different from the parent. You will have to call `unpackChildNodesUsedRoutes` to get the routes for the children.

Reply:
I had seen this unpackCHildNodesUsedRoutes but it can return a list of routes. What should I do in this case ?

Reply:
It returns a map. You need to get the used nodes of the child via its id.

Reply:
I sync up with navin offline and I think I got it.